commit.py

```diff
@@ -77,7 +77,14 @@ def get(self, message_hash=None):
         if not message_hash:
             message_hash = random.choice(messages.keys())
         elif message_hash not in messages:
-            raise tornado.web.HTTPError(404)
+            state = random.getstate()
+            try:
+                random.seed(int(message_hash, 16))
+                message_hash = random.choice(messages.keys())
+            except:
+                raise tornado.web.HTTPError(404)
+            finally:
+                random.setstate(state)
 
         message = fill_line(messages[message_hash])
```

Comment:
You probably want to pass in the message hash to the `fill_line` function so that the selection of names and numbers (if applicable) uses the same seed.